testutil/integration/evmos/factory/factory.go

```diff
@@ -81,21 +92,25 @@ func New(
 	}
 }
 
-// createMsgEthereumTx creates a new MsgEthereumTx with the provided arguments.
-// If any of the arguments are not provided, they will be populated with default values.
-func (tf *IntegrationTxFactory) createMsgEthereumTx(
-	privKey cryptotypes.PrivKey,
-	txArgs evmtypes.EvmTxArgs,
-) (evmtypes.MsgEthereumTx, error) {
-	fromAddr := common.BytesToAddress(privKey.PubKey().Address().Bytes())
-	// Fill TxArgs with default values
-	txArgs, err := tf.populateEvmTxArgs(fromAddr, txArgs)
-	if err != nil {
-		return evmtypes.MsgEthereumTx{}, errorsmod.Wrap(err, "failed to populate tx args")
+// GetEvmTransactionResponseFromTxResult returns the MsgEthereumTxResponse from the provided txResult.
+func (tf *IntegrationTxFactory) GetEvmTransactionResponseFromTxResult(
```

Comment:
nit: This is a pretty hefty method name 😅 
```suggestion
func (tf *IntegrationTxFactory) GetEvmTxResponseFromTxResult(
```